test/e2e/apimachinery/garbage_collector.go

```diff
@@ -1101,14 +1101,14 @@ var _ = SIGDescribe("Garbage collector", func() {
 			framework.Failf("timeout in waiting for the owner to be deleted: %v", err)
 		}
 
-		// Wait 30s and ensure the dependent is not deleted.
-		ginkgo.By("wait for 30 seconds to see if the garbage collector mistakenly deletes the dependent crd")
-		if err := wait.PollWithContext(ctx, 5*time.Second, 30*time.Second+gcInformerResyncRetryTimeout, func(ctx context.Context) (bool, error) {
-			_, err := resourceClient.Get(ctx, dependentName, metav1.GetOptions{})
-			return false, err
-		}); err != nil && !wait.Interrupted(err) {
-			framework.Failf("failed to ensure the dependent is not deleted: %v", err)
-		}
+		// Wait 90s and ensure the dependent is not deleted.
+		ginkgo.By("wait for 90 seconds to see if the garbage collector mistakenly deletes the dependent crd")
```

Comment:
Who do you know that it's 90 seconds? Better don't make assumptions about the value of `gcInformerResyncRetryTimeout`:
```suggestion
                timeout := 30*time.Second + gcInformerResyncRetryTimeout
		ginkgo.By(fmt.Sprintf("wait for %s to see if the garbage collector mistakenly deletes the dependent crd"), timeout))
```

Reply:
Yes, you are absolutely right. Thank you very much for your advice.

Reply:
It seems that a match for error type is needed here？
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/121583/pull-kubernetes-e2e-kind/1719286645296992256